addons/calendar/static/src/activity/activity_patch.js

```diff
@@ -18,7 +18,8 @@ patch(Activity.prototype, {
     async unlink() {
         if (this.props.data.calendar_event_id) {
             await this.orm.call("mail.activity", "unlink_w_meeting", [[this.props.data.id]]);
-            this.props.onUpdate();
+            this.activityService.delete(this.props.data);
```

Comment:
Note: in the super call, the call to the service delete is actually done before the await rpc.
I'm not sure how important that is.

Reply:
Also, the thread is saved before doing the delete, because it might become undefined in the meantime.